src/ot_orchestration/operators/manifest_operators.py

```diff
@@ -0,0 +1,284 @@
+from airflow.models.baseoperator import BaseOperator
+from typing import Sequence
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.providers.google.cloud.operators.cloud_batch import (
+    CloudBatchSubmitJobOperator,
+)
+from google.cloud.batch_v1 import Environment
+import re
+from ot_orchestration.utils.manifest import extract_study_id_from_path
+from ot_orchestration.utils.batch import create_batch_job, create_task_spec
+from ot_orchestration.types import Manifest_Object
+import logging
+from ot_orchestration.utils.path import IOManager
+from airflow.exceptions import AirflowSkipException
+
+
+class ManifestGenerateOperator(BaseOperator):
+    """Generate manifest dictionary for each summary statistics."""
+
+    template_fields: Sequence[str] = (
+        "raw_sumstat_path_pattern",
+        "staging_manifest_path_pattern",
+        "harmonised_prefix",
+        "qc_prefix",
+    )
+
+    def __init__(
+        self,
+        *,
+        raw_sumstat_path_pattern: str,
+        staging_manifest_path_pattern: str,
+        harmonised_prefix: str = "harmonised",
+        qc_prefix: str = "qc",
+        gcp_conn_id: str = "google_cloud_default",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.raw_sumstat_path_pattern = raw_sumstat_path_pattern
+        self.staging_manifest_path_pattern = staging_manifest_path_pattern
+        self.harmonised_prefix = harmonised_prefix
+        self.qc_prefix = qc_prefix
+        self.gcp_conn_id = gcp_conn_id
+
+    def execute(self, **_) -> list[Manifest_Object]:
+        """Execute the operator.
+
+        Raises:
+            ValueError: when incorrect glob is defined
+            ValueError: when the glob protocol is not gs
+
+        Returns:
+            dict[str, str]: list of manifests
+        """
+        # this regex pattern can be utilized for any path or uri glob pattern
+        pattern = r"^((?P<protocol>.*)://)?(?P<root>[(\w)-]+)/(?P<prefix>([(\w)-/])+?)/(?P<matchglob>[(\w)-*]+.*){1}"
+        compiled_pattern = re.compile(pattern)
+
+        globs = {
+            "raw_sumstat": self.raw_sumstat_path_pattern,
+            "manifest": self.staging_manifest_path_pattern,
+        }
+
+        results = {}
+        for key, glob in globs.items():
+            _match = compiled_pattern.match(glob)
+            if _match is None:
+                raise ValueError("Incorrect glob pattern %s", glob)
+            protocol = _match.group("protocol")
+            root = _match.group("root")
+            prefix = _match.group("prefix")
+            matchglob = _match.group("matchglob")
+
+            if protocol != "gs":
+                raise NotImplementedError(
+                    "Listing objects from path with %s protocol is not implemented",
+                    protocol,
+                )
+            logging.info(
+                "Listing files at %s/%s with match glob %s", root, prefix, matchglob
+            )
```

Comment:
```suggestion
            self.log.info(
                "Listing files at %s/%s with match glob %s", root, prefix, matchglob
            )
```

You can use `self.log` in Operators subclassing `BaseOperator`, as it initializes its own logger inside with Airflow custom settings.